apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerExceptionTests.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright 2021 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Grpc.Core;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public class SpannerExceptionTests
+    {
+        [Fact]
+        public void IsRetryable()
```

Comment:
I'd suggest this would be clearer using some Theory tests with InlineTheoryData - you probably need two tests, one taking an ErrorCode, a string and an expected result, and the other taking an ErrorCode, a StatusCode, a string and an expected result.

Reply:
I've updated the test to a data-driven test, and that does indeed look tidier.

Please hold off on merging this one, as there's some discussion going on whether we should really retry this.

Reply:
> Please hold off on merging this one, as there's some discussion going on whether we should really retry this.

Ack.

Reply:
Any more news on whether we should actually retry?

Reply:
Yes and no... We should retry in most cases, but we're awaiting confirmation for exactly what cases we should not retry (or when we need to do additional checks to determine whether it is safe to retry)